src/Metadata/Resource/Factory/OperationResourceMetadataFactory.php

```diff
@@ -123,6 +124,13 @@ private function normalize(bool $collection, ResourceMetadata $resourceMetadata,
                 $supported ? $operation['method'] = $upperOperationName : $operation['route_name'] = $operationName;
             }
 
+            if (strtoupper($operation['method']) !== $operation['method']) {
+                $message = sprintf('The method "%s" of operation "%s" should be in uppercase.', $operation['method'], $operationName);
+
+                // See DelegatingLoader::load to understand why we use a LoaderLoadException
```

Comment:
Sorry, I think it's very bad for our factory to throw this exception. It's completely unexpected and will catch any consumer of this code off-guard. We should find a better way if we want to do this.